p256/src/ecdsa.rs

```diff
@@ -107,6 +105,24 @@ mod tests {
     use super::*;
     use crate::test_vectors::ecdsa::ECDSA_TEST_VECTORS;
 
+    #[cfg(feature = "rand")]
+    use {crate::BlindedScalar, elliptic_curve::rand_core::OsRng};
+
     ecdsa::new_signing_test!(ECDSA_TEST_VECTORS);
     ecdsa::new_verification_test!(ECDSA_TEST_VECTORS);
+
+    #[cfg(feature = "rand")]
+    #[test]
+    fn scalar_blinding() {
+        let vector = &ECDSA_TEST_VECTORS[0];
+        let d = Scalar::from_bytes(vector.d.try_into().unwrap()).unwrap();
+        let k = Scalar::from_bytes(vector.k.try_into().unwrap()).unwrap();
+        let k_blinded = BlindedScalar::new(k, &mut OsRng);
+        let sig = d
+            .try_sign_prehashed(&k_blinded, GenericArray::from_slice(vector.m))
+            .unwrap();
+
+        assert_eq!(vector.r, sig.r().as_slice());
+        assert_eq!(vector.s, sig.s().as_slice());
+    }
```

Comment:
FYI @nickray, here's the proposed alternative for `masking_scalar`.

I had it take an RNG argument directly, but we could also add an API to construct one from two `Scalar` values you generate however you want (perhaps that could be `new`, and the RNG-based one could be `generate`)